### ISSUE TITLE
Get Operating System name for CLI and Desktop Apps

### ISSUE THREAD

**Issue by FanJups** [osscameroon/js-generator#196]
It will be really useful to know the OS used by the user in order to download the version matching the OS. 

As an example, [Node Js](https://nodejs.org/en/) is doing something great, they provide the option matching the OS without the need for the user to choose amongst other OS versions. Now I'am using Windows then they show me directly the windows versions. We should also let the user to choose amongst versions, maybe an idea could be that the user is using Windows but wants to download the Linux or Mac version to send to someone else or use in virtual machine.... 

![image](https://user-images.githubusercontent.com/25267726/205070258-8059e7eb-3f38-48fa-bca7-4e1e8b521230.png)

To sum up, by default, we show the version matching the OS but we also let the user to choose.

Some useful links:

https://www.google.com/search?q=java+os.name&rlz=1C1FCXM_pt-PTPT1032PT1032&oq=Java+os.name&aqs=chrome.0.0i512l6j0i22i30l4.12235j1j7&sourceid=chrome&ie=UTF-8

https://mkyong.com/java/how-to-detect-os-in-java-systemgetpropertyosname/

https://stackoverflow.com/questions/31909107/javas-os-name-for-windows-10

https://www.baeldung.com/java-detect-os

**Comment by FanJups** [comment]
User Agent will help us to do that https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
